### ISSUE TITLE
Event Hub Consumer Group name does not allow underscores

### ISSUE THREAD

**Issue by ducas** [terraform-providers/terraform-provider-azurerm#1800]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.11.7

### Affected Resource(s)

* azurerm_eventhub_consumer_group

### Terraform Configuration Files

```hcl
terraform {}

provider "azurerm" {
  version = "1.13.0"
}

resource "azurerm_eventhub_consumer_group" "test_consumer" {
  name                = "hello_world"
  namespace_name      = "my_ns"
  eventhub_name       = "my_hub"
  resource_group_name = "my_rg"
}
```

### Debug Output

```
2018/08/21 13:14:35 [INFO] Terraform version: 0.11.7
2018/08/21 13:14:35 [INFO] Go runtime version: go1.10.1
2018/08/21 13:14:35 [INFO] CLI args: []string{"/usr/local/Cellar/terraform/0.11.7/bin/terraform", "plan"}
2018/08/21 13:14:35 [DEBUG] Attempting to open CLI config file: /Users/ducas/.terraformrc
2018/08/21 13:14:35 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2018/08/21 13:14:35 [INFO] CLI command args: []string{"plan"}
2018/08/21 13:14:35 [INFO] command: empty backend config, returning nil
2018/08/21 13:14:35 [DEBUG] command: no data state file found for backend config
2018/08/21 13:14:35 [DEBUG] New state was assigned lineage "7f12f3f5-2edf-1a5c-7dec-667e5ad5e1a9"
2018/08/21 13:14:35 [INFO] command: backend initialized: <nil>
2018/08/21 13:14:35 [DEBUG] checking for provider in "."
2018/08/21 13:14:35 [DEBUG] checking for provider in "/usr/local/Cellar/terraform/0.11.7/bin"
2018/08/21 13:14:35 [DEBUG] checking for provider in ".terraform/plugins/darwin_amd64"
2018/08/21 13:14:35 [DEBUG] found provider "terraform-provider-azurerm_v1.13.0_x4"
2018/08/21 13:14:35 [DEBUG] found valid plugin: "azurerm", "1.13.0", "/Users/ducas/temp/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.13.0_x4"
2018/08/21 13:14:35 [DEBUG] checking for provisioner in "."
2018/08/21 13:14:35 [DEBUG] checking for provisioner in "/usr/local/Cellar/terraform/0.11.7/bin"
2018/08/21 13:14:35 [DEBUG] checking for provisioner in ".terraform/plugins/darwin_amd64"
2018/08/21 13:14:35 [INFO] command: backend <nil> is not enhanced, wrapping in local
2018/08/21 13:14:35 [INFO] backend/local: starting Plan operation
2018/08/21 13:14:35 [DEBUG] New state was assigned lineage "ab7dbdfc-cdf7-825d-3faa-4178c8779956"
2018/08/21 13:14:35 [INFO] terraform: building graph: GraphTypeInput
2018/08/21 13:14:35 [DEBUG] Resource state not found for "azurerm_eventhub_consumer_group.test_consumer": azurerm_eventhub_consumer_group.test_consumer
2018/08/21 13:14:35 [TRACE] Graph after step *terraform.AttachStateTransformer:

azurerm_eventhub_consumer_group.test_consumer - *terraform.NodeAbstractResource
2018/08/21 13:14:35 [DEBUG] ReferenceTransformer: "azurerm_eventhub_consumer_group.test_consumer" references: []
2018/08/21 13:14:35 [DEBUG] ReferenceTransformer: "provider.azurerm" references: []
2018/08/21 13:14:35 [DEBUG] Starting graph walk: walkInput
2018-08-21T13:14:35.530+1000 [DEBUG] plugin: starting plugin: path=/Users/ducas/temp/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.13.0_x4 args=[/Users/ducas/temp/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.13.0_x4]
2018-08-21T13:14:35.537+1000 [DEBUG] plugin: waiting for RPC address: path=/Users/ducas/temp/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.13.0_x4
2018-08-21T13:14:35.545+1000 [DEBUG] plugin.terraform-provider-azurerm_v1.13.0_x4: plugin address: timestamp=2018-08-21T13:14:35.545+1000 address=/var/folders/05/jzcw1_q924d3mxf7fg_gl_3m0000gn/T/plugin777199740 network=unix
2018/08/21 13:14:35 [INFO] terraform: building graph: GraphTypeValidate
2018/08/21 13:14:35 [DEBUG] Resource state not found for "azurerm_eventhub_consumer_group.test_consumer": azurerm_eventhub_consumer_group.test_consumer
2018/08/21 13:14:35 [DEBUG] ReferenceTransformer: "azurerm_eventhub_consumer_group.test_consumer" references: []
2018/08/21 13:14:35 [DEBUG] ReferenceTransformer: "provider.azurerm" references: []
2018/08/21 13:14:35 [TRACE] Graph after step *terraform.ReferenceTransformer:

azurerm_eventhub_consumer_group.test_consumer - *terraform.NodeValidatableResource
  provider.azurerm - *terraform.NodeApplyableProvider
provider.azurerm - *terraform.NodeApplyableProvider
2018/08/21 13:14:35 [DEBUG] Starting graph walk: walkValidate
2018/08/21 13:14:35 [DEBUG] Resource state not found for "azurerm_eventhub_consumer_group.test_consumer": azurerm_eventhub_consumer_group.test_consumer
2018/08/21 13:14:35 [DEBUG] ReferenceTransformer: "azurerm_eventhub_consumer_group.test_consumer" references: []
2018/08/21 13:14:35 [ERROR] root: eval: *terraform.EvalValidateResource, err: Warnings: []. Errors: [invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.) invalid value for namespace_name (The namespace name can contain only letters, numbers, underscores and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)]
2018/08/21 13:14:35 [ERROR] root: eval: *terraform.EvalSequence, err: Warnings: []. Errors: [invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.) invalid value for namespace_name (The namespace name can contain only letters, numbers, underscores and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)]
2018/08/21 13:14:35 [TRACE] [walkValidate] Exiting eval tree: azurerm_eventhub_consumer_group.test_consumer

2018/08/21 13:14:35 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-08-21T13:14:35.571+1000 [DEBUG] plugin.terraform-provider-azurerm_v1.13.0_x4: 2018/08/21 13:14:35 [ERR] plugin: plugin server: accept unix /var/folders/05/jzcw1_q924d3mxf7fg_gl_3m0000gn/T/plugin777199740: use of closed network connection
Error: azurerm_eventhub_consumer_group.test_consumer: invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)



Error: azurerm_eventhub_consumer_group.test_consumer: invalid value for namespace_name (The namespace name can contain only letters, numbers, underscores and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)


2018-08-21T13:14:35.573+1000 [DEBUG] plugin: plugin process exited: path=/Users/ducas/temp/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.13.0_x4
```

### Panic Output

NA

### Expected Behavior

Creates consumer

### Actual Behavior

```
Error: azurerm_eventhub_consumer_group.test_consumer: invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)
```

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`

### Important Factoids

Works on version 1.9.0. 

### References

* NA

**Comment by tombuildsstuff** [comment]
hey @ducas

Just to let you know that this has been released in v1.14.0 of the AzureRM Provider which is now available: https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.14.0/CHANGELOG.md

Thanks!

**Comment by ducas** [comment]
Great stuff!
